onnxscript/function_libs/torch_aten/ops/core.py

```diff
@@ -269,10 +269,32 @@ def aten_angle(self: TensorType) -> TensorType:
     raise NotImplementedError()
 
 
-def aten_any(self: TensorType) -> TensorType:
+@torch_op("aten::any", trace_only=True)
+def aten_any(self: TTensor, dim: Optional[int] = None, keepdim: bool = True) -> BOOL:
     """any(Tensor self) -> Tensor"""
 
-    raise NotImplementedError()
+    minus_1 = op.Constant(value_ints=[-1])
+    self_rank = op.Size(op.Shape(self))
+    if self_rank == 0:
+        self = op.Reshape(self, minus_1)
+
+    zero = op.Constant(value_float=0.0)
+    result = op.Not(op.Equal(self, zero))
```

Comment:
Why not just cast to BOOL?

Reply:
Address by https://github.com/microsoft/onnx-script/issues/498 @xiaowuhu i would link the new prs here for reference